src/main/java/com/amazon/ion/impl/macro/EExpressionArgsReader.java

```diff
@@ -231,15 +231,16 @@ private void readStreamAsExpressionGroup(
      * @param expressions receives the expressions as they are materialized.
      */
     protected void readValueAsExpression(boolean isImplicitRest, List<Expression.EExpressionBodyExpression> expressions) {
-        if (isMacroInvocation()) {
+        if (isImplicitRest && !isContainerAnExpressionGroup()) {
+            readStreamAsExpressionGroup(expressions);
+            return;
+        } else if (isMacroInvocation()) {
```

Comment:
🗺️ Had to change the order of some checks in this method so that it would correctly capture rest args when the first of the rest args was a macro invocation.